app/src/main/java/org/oppia/android/app/home/WelcomeViewModel.kt

```diff
@@ -13,14 +13,18 @@ class WelcomeViewModel(
   dateTimeUtil: DateTimeUtil
 ) : HomeItemViewModel() {
 
-  /** Text [String] to greet the learner and display on-screen when launching the home activity. */
+  /** Text [String] to greet the learner. */
   val greeting: String = dateTimeUtil.getGreetingMessage()
 
   /**
-   * Returns the user-readable portion of the welcome screen greeting that contains the user's name.
+   *  Returns the string which contains greeting message with user's name and
+   *  display on-screen when launching the home activity.
    */
-  fun computeProfileNameText(): String {
-    return resourceHandler.getStringInLocaleWithWrapping(R.string.welcome_profile_name, profileName)
+  fun welcomeText(): String {
+    val profName = resourceHandler.getStringInLocaleWithWrapping(
+      R.string.welcome_profile_name, profileName
+    )
+    return "$greeting $profName"
   }
```

Comment:
You could improve the string concatenation by doing the following:
Modify the welcome string to expect two arguments, the greeting and the profile name - e.g.
` <string name="welcome_profile_name">%s %s!</string>`.

Then pass the computed greeting text as an argument
```suggestion
  fun computeWelcomeText(): String  {
       return resourceHandler.getStringInLocaleWithWrapping(R.string.welcome_profile_name, greeting, profileName)
  }
```

or 

```suggestion
  fun computeWelcomeText(): String = resourceHandler.getStringInLocaleWithWrapping(R.string.welcome_profile_name, greeting, profileName)
```

Reply:
@ShubhadeepKarmakar this comment has not been addressed.